stock_no_negative/models/stock_location.py

```diff
@@ -12,3 +12,7 @@ class StockLocation(models.Model):
         string='Allow Negative Stock',
         help="Allow negative stock levels for the stockable products "
         "attached to this location.")
+
+    @api.multi
+    def _is_negative_allowed(self):
+        return self.allow_negative_stock or self.usage not in ['internal', 'transit']
```

Comment:
I would have added self.ensure_one() here or would have implemented a filtered() on a real multi recordset

Reply:
Done, added `ensure_one` to be [consistent with Odoo](https://github.com/odoo/odoo/blob/f845124535e2c882656bf0ff212dcd539d004ae3/addons/stock_account/models/stock.py#L227)

Reply:
FYI it doesn't have ensure_one

Is it fine?

Reply:
@moylop260 You are absolutely right! It seems I forgot to push :confused: